src/lib/VirtualScroller.js

```diff
@@ -0,0 +1,183 @@
+class VirtualScroller {
+    /**
+     * [constructor]
+     *
+     * @param {HTMLElement} container - The HTMLElement that will contain the virtual scroller
+     * @return {VirtualScroller} Instance of VirtualScroller
+     */
+    constructor(container) {
+        this.container = container;
+
+        this.previousScrollTop = 0;
+
+        this.onScrollHandler = this.onScrollHandler.bind(this);
+        this.renderItems = this.renderItems.bind(this);
+    }
+
+    /**
+     * Destroys the virtual scroller
+     *
+     * @return {void}
+     */
+    destroy() {
+        if (this.containerEl) {
+            this.containerEl.remove();
+        }
+
+        this.containerEl = null;
+        this.contentEl = null;
+    }
+
+    /**
+     * Initializes the virtual scroller
+     *
+     * @param {Object} config - The config
+     * @return {void}
+     */
+    init(config) {
+        if (!config.totalItems || !config.itemHeight) {
```

Comment:
Are there other required properties? It looks like `containerHeight` and `renderItemFn` are also needed.